### ISSUE TITLE
registerBlockStyle for `core/column` results in map.undefined when user's mouse enters the style area of components-popover__content

### ISSUE THREAD

**Issue by fuzzywalrus** [WordPress/gutenberg#30588]
## Description

When using registerBlockStyle to enable custom styles in a custom block style plugin will work from the side bar. However, if the user interacts with it using the component-popover, it'll cause a `map of undefined` error:

Here's the partial error log:
```blocks.js?ver=9ed25ffa009c799f99a4340915b6dc6a:7852 Uncaught TypeError: Cannot read property 'map' of undefined
    at cloneBlock (blocks.js?ver=9ed25ffa009c799f99a4340915b6dc6a:7852)
    at BlockStylesMenu (block-editor.js?ver=7bc2f41a2f2e0da426f12f29b2139cae:23825)
    at renderWithHooks (react-dom.js?ver=16.13.1:14938)
    at updateFunctionComponent (react-dom.js?ver=16.13.1:17169).....
```

```
react_devtools_backend.js:2430 The above error occurred in the <BlockStylesMenu> component:
    in BlockStylesMenu (created by Dropdown)
    in div (created by Dropdown)
    in div (created by NavigableContainer)
    in NavigableContainer (created by ForwardRef(NavigableContainer))
    in ForwardRef(NavigableContainer) (created by ForwardRef(NavigableMenu))
    in ForwardRef(NavigableMenu) (created by Dropdown)
    in div (created by Popover)...
```

## Step-by-step reproduction instructions

1. Register a custom style for `core/column`
2. Try to apply custom styles from the inline popover editor

## Expected behaviour
No crash.

## Actual behavior
"The editor has encountered an unexpected error."

This same issue happened with `<InnerBlocks>` on a custom plugin I wrote but i was able to fix it using API 2 and using `useBlockProps()` but seeing as this is a core plugin I'm not really sure how to trouble shot it. 

## Screenshots or screen recording (optional)
![component popover](http://www.greggant.com/randomstuff/photodump/crash.png)

## Code snippet (optional)

I've set up a very basic plugin to enqueue the css/js:
index.php
```
function block_styles_enqueue_javascript() {
	wp_enqueue_script( 'block-styles-script',
		plugins_url( 'block.js', __FILE__ ),
		array( 'wp-blocks')
	);
}
add_action( 'enqueue_block_editor_assets', 'block_styles_enqueue_javascript' );
function block_styles_enqueue_stylesheet() {
	wp_enqueue_style( 'block-styles-stylesheet',
		plugins_url( 'style.css', __FILE__ )
	);
}
add_action( 'enqueue_block_assets', 'block_styles_enqueue_stylesheet' );
```

block.js 
```wp.blocks.registerBlockStyle( 'core/column', {
	name: 'column-max-width-md',
	label: 'Medium Width Column'
});
wp.blocks.registerBlockStyle( 'core/column', {
	name: 'column-left-md',
	label: 'Left Aligned Medium Column'
});
```

## WordPress information
- WordPress version: 5.7
- Gutenberg version:  (unclear where you check this)
- Are all plugins except Gutenberg deactivated?  Yes (except for custom plugin for styles)
- Are you using a default theme (e.g. Twenty Twenty-One)? No

## Device information
- Device: Desktops, BrowserStack
- Operating system: macOS 10.15, macOS 10.16
- Browser: FireFox, Chrome, Edge, Safari

**Comment by gziolo** [comment]
I can't reproduce it with the most recent version of the Gutenberg plugin. @ntsekouras, isn't it something we fixed after WP 5.7?

**Comment by ntsekouras** [comment]
I can't reproduce either and it seems to have been fixed by: https://github.com/WordPress/gutenberg/pull/29894.

>isn't it something we fixed after WP 5.7?

Yes.

>Gutenberg version: (unclear where you check this)

@fuzzywalrus This means if you have enabled the Gutenberg plugin in your site. So you have to go the `Plugins` page and see.

**Comment by gziolo** [comment]
It looks like I fixed the issue. Thanks Nik for the link 😄
